### PR TITLE
Issue 4153 - Color picker

### LINKED ISSUE #4153: Custom Tickets: Groups - Colour picker control
### Description
A new colour picker will be necessary for the groups feature in custom tickes

### Goals
- [ ] To have a more usefule/pretty colour picker

### Tasks
- [ ] Create the control (should be seen as an input for react-hook forms to pick up seamlessly)
- [ ] Create the predefined colours and hex editor panel
- [ ] Create the picking colour bit 

### Related Resources
[MockUp](https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=3251-32180&t=CCepwuyWCdOc7GP7-0)

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #4153

#### Description
Added the color picker as a storybook component

#### Test cases
Navigate to the storybook section and play around with colours, opacity and the colour picker

### CODE REVIEW

File: frontend/src/v5/ui/controls/inputs/colorPicker/colorPickerPalette/colorPickerPalette.styles.ts
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/ui/controls/inputs/colorPicker/colorPickerPalette/colorPickerPalette.styles.ts:1]: I fixed both, but for the leading 0(s) problem I resorted to a slight hack, i.e. I converted the value to a string before passing it to the NumberField.
Due to a bug in React where the numbers "004" and "4" are the same, the component does not update correctly.

I wonder though, is it fine to have this hack?

**Review by The-Daniel — DISMISSED** [review]

**Comment by sanmont3drepo** [comment]
There is an inconsistency when selecting the colour. When I click out the of the display menu the input becomes unselected.
But when I click on the x it still selected. Both actions changes the colour

selected

**Comment by sanmont3drepo** [comment]
Changing the colours in storybook doesnt work or Im doing something wrong?
image

**Comment by sanmont3drepo** [comment]
Changing the colours in storybook doesnt work or Im doing something wrong?
image

File: frontend/src/v5/ui/components/viewer/cards/tickets/pinDetails/pinDetails.component.tsx [hunk 1]
```diff
@@ -83,34 +84,32 @@ export const PinDetails = ({ value, label, onChange, onBlur, required, error, he
...
-				<PinName onClick={onClickEditPin} required={required}>
-					{label}
-				</PinName>
-				<PinActions>
-					{editMode && (
-						<SettingLocationText onClick={cancelEdit}>
-							<FormattedMessage id="tickets.pin.selectLocation" defaultMessage="Select new location on model" /> <CrossIcon />
-						</SettingLocationText>
-					)}
+		<CustomInput required={required} selected={editMode} error={error} disabled={disabled}>
```
Comment (sanmont3drepo) [inline frontend/src/v5/ui/components/viewer/cards/tickets/pinDetails/pinDetails.component.tsx:87]: Change this name to inputcontainer

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

**Comment by Amantini1997** [comment]
> Changing the colours in storybook doesnt work or Im doing something wrong? image

It should be fixed now. In case it still seems to not be working, hit the **storybook refresh** button after changing the args
image

**Review by sanmont3drepo — APPROVED** [review]

**Pull request merged**